src/home_work_14_1.ts

```diff
@@ -0,0 +1,96 @@
+interface Note {
+  id: string;
+  title: string;
+  content: string;
+  createdAt: Date;
+  updatedAt: Date;
+  status: "completed" | "pending";
+  confirmEdit: boolean;
+}
+
+class TodoList {
+  notes: Note[];
+
+  constructor() {
+    this.notes = [];
+  }
+
+  addNote(note: Note) {
+    if (note.title.trim() === "" || note.content.trim() === "") {
+      throw new Error("Нотатка не може бути порожньою");
+    }
+    this.notes.push(note);
+  }
+
+  deleteNote(id: string) {
+    const index = this.notes.findIndex((note) => note.id === id);
+    if (index !== -1) {
+      this.notes.splice(index, 1);
+    }
+  }
+
+  editNote(id: string, newContent: string) {
+    const note = this.notes.find((note) => note.id === id);
+    if (note) {
+      if (note.confirmEdit) {
+        throw new Error("Редагування цієї нотатки вимагає підтвердження");
+      }
+      note.content = newContent;
+      note.updatedAt = new Date();
+    }
+  }
+
+  getNote(id: string) {
+    return this.notes.find((note) => note.id === id);
+  }
+
+  getAllNotes() {
+    return this.notes;
+  }
+
+  markAsCompleted(id: string) {
+    const note = this.notes.find((note) => note.id === id);
+    if (note) {
+      note.status = "completed";
```

Comment:
It would be better to create an enum for these cases. Using string values in code is not a best practice.